### ISSUE TITLE
future::Stream::collect() breaks autocompletion & makes false-positive 'Unresolved reference' 

### ISSUE THREAD

**Issue by bool-rus** [intellij-rust/intellij-rust#3111]
## Environment

* **Intellij-Rust plugin version:**
v0.2.0.2111-183

* **Rust toolchain version:**
cargo 1.30.0 (a1a4ad372 2018-11-02)
rustc 1.30.1 (1433507eb 2018-11-07)
rustup 1.15.0 (f0a3d9a84 2018-11-27)
racer 2.0.14

* **IDE name and version:**
IDEA 2018.3 CE
* **Operating system:**
MacOS: Darwin Kernel Version 18.2.0

## Problem description
After using the *collect* method of **futures :: future :: Stream** method
I cannot use autocomplete because it does not work for this variable. In addition, a false prompt 'Unresolved reference' appears for all methods applied to this variable.

## Steps to reproduce
there is right code below:

    use futures::Stream;
    use futures::Future;
    fn example_stream<S: Stream>(s: S) -> impl Future<Item=usize, Error=S::Error> {
        s.collect().map(|v|v.len())
    }

autocomplete:
<img width="608" alt="image" src="https://user-images.githubusercontent.com/1818107/49427810-60349480-f7b5-11e8-811e-f3b658aa6ee2.png">
code tip (false positive):
<img width="595" alt="image" src="https://user-images.githubusercontent.com/1818107/49434713-d93be800-f7c5-11e8-85d5-3512f574e683.png">

## Workaround:

    use futures::Stream;
    use futures::Future;
    
    fn collect<S:Stream>(s:S) -> impl Future<Item=Vec<S::Item>, Error=S::Error> {
        s.collect()
    }
    
    fn example_stream<S: Stream>(s: S) -> impl Future<Item=usize, Error=S::Error> {
        collect(s).map(|v|v.len())
    }

well, it's worked:
<img width="588" alt="image" src="https://user-images.githubusercontent.com/1818107/49436008-063dca00-f7c9-11e8-93dc-7a0336f2ce25.png">

**Comment by chuxi** [comment]
actix_web multipart example has the same problem. 

https://github.com/actix/examples/blob/master/multipart/src/main.rs

**Comment by Kobzol** [comment]
This seems to work with the current version of the plugin and `futures` `0.3`.
```rust
use futures::{Stream, StreamExt};
use futures::Future;

fn example_stream<S: Stream>(s: S) -> impl Future<Item=usize, Error=S::Error> {
    s.collect().map(|v|v.len())
}
```
